runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2596,6 +2612,60 @@ void TR::CompilationInfo::resumeCompilationThread()
    releaseCompMonitor(vmThread);
    }
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+void TR::CompilationInfo::prepareForCheckpoint()
+   {
+   J9JavaVM   *vm       = _jitConfig->javaVM;
+   J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
+
+   {
+   OMR::CriticalSection suspendCompThreadsForCheckpoint(getCompilationMonitor());
+
+   /* Set the checkpoint in progress flag. */
+   setCheckpointInProgress();
+
+   /* Inform compilation threads to suspend themselves. The compilation
+    * threads will complete any in-progress compilations first.
+    */
+   bool purgeMethodQueue = false;
+   suspendCompilationThread(purgeMethodQueue);
+   }
+
+   /* Wait until all compilation threads are suspended. */
```

Comment:
Hm, I may need to check the checkpointInRestore flag in any locations that try to resume threads. Theoretically after the comp thread suspends itself, it could be resumed in which case we could still have the problem where a comp thread is active across a checkpoint/restore operation.

Reply:
Addressed in https://github.com/eclipse-openj9/openj9/pull/14331/commits/e23a2dce8c87b6e0b6ee7339a98ed5b54c54118c